onkelhoy/web-components
# add: global testing

note: this still requires github actions settings to be configured manually
closes: #33
changelog:
- fix: not including lib folder

- fix: server file handle updated to work with paths without ending slash (deep paths)

- patch: server

- fix: theme packages updated to have lib folder to stay consisent

- fix: global build script (accidentally got change to "lib")

- fix: ci-pipeline publish upgrade to v4

- change: theme includes layer info in config for consistency - helps global test

- fix: server get-url

- change: test run globally

- update: iterate package script splitted out from publish

- add: global test script

- clean: snapshots + correcting so tests can run in module mode

- fix: adding open mode to start script (server)

- shape: contributing hidden into .github folder - should have a deeper look into this one day with lots of different files

- add: server open mode introduced

- patch: server (new open mode)

- add: global test working

- fix: tests should not route a hardcore route but rather a soft-route

- fix: build script cleaner and better + weird _utils/packages removed

- add: manual trigger of workflow

- add: manual trigger to publish also

- add: test on specific branch

- fix: switch to bash

- test: switch all to bash

- patch: server

- fix: bash ans sh misstake

- patch: server

- add: playwright install step

- add: papit-server install to workflow

- wip: papit-server and playwright into 2 steps + playwright cache

- fix: increase snapshot pixel diff ratio

- clean: no list of result folders (old debug)

- add: test exit code notice

- fix: update snapshot + icon default font

- add: snapshots - good indication as git cannot pick up on image changes

- change: snapshot prerun

- change: not running with update snapshots each time as test will do if not there regardless

- fix: snapshot file name

- update: generate template unit and snapshot

- fix: safer print and git add

- fix: core package config update name

- clean: workflow file corrected and now works with cache updats

- format: yml no string on false ?

- fix: rootdir needs to be exported for bash-dependency-orer

- fix: cleanup scripts and put build into its own place

- fix: publish need not to build anymore as its going to be called by CI (normally) and this already builds

- fix: correcting output-step

- fix: skip-action check + cleanup
